stackit/internal/services/secretsmanager/instance/resource_test.go

```diff
@@ -134,3 +176,296 @@ func TestToCreatePayload(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateACLs(t *testing.T) {
+	// This is the response used when getting all ACLs currently, across all tests
+	getAllACLsResp := secretsmanager.AclList{
+		Acls: &[]secretsmanager.Acl{
+			{
+				Cidr: utils.Ptr("acl-1"),
+				Id:   utils.Ptr("id-acl-1"),
+			},
+			{
+				Cidr: utils.Ptr("acl-2"),
+				Id:   utils.Ptr("id-acl-2"),
+			},
+			{
+				Cidr: utils.Ptr("acl-3"),
+				Id:   utils.Ptr("id-acl-3"),
+			},
+		},
+	}
+	getAllACLsRespBytes, err := json.Marshal(getAllACLsResp)
+	if err != nil {
+		t.Fatalf("Failed to marshal get all ACLs response: %v", err)
+	}
+
+	// This is the response used whenever an API returns a failure response
+	failureRespBytes := []byte("{\"message\": \"Something bad happened\"")
+
+	tests := []struct {
+		description        string
+		acls               []string
+		getAllACLsFails    bool
+		createACLFails     bool
+		deleteACLFails     bool
+		isValid            bool
+		expectedACLsStates map[string]bool // Keys are CIDR; value is true if CIDR should exist at the end, false if should be deleted
+	}{
+		{
+			description: "no_changes",
+			acls:        []string{"acl-3", "acl-2", "acl-1"},
+			expectedACLsStates: map[string]bool{
+				"acl-1": true,
+				"acl-2": true,
+				"acl-3": true,
+			},
+			isValid: true,
+		},
+		{
+			description: "create_acl",
+			acls:        []string{"acl-1", "acl-2", "acl-3", "acl-4"},
```

Comment:
Does Terraform already handle the case if the user adds two repeated elements to the set?

Reply:
Good catch! Turns out no, `types.Set` doesn't preserve order but preserves repetitions apparently. Running `ACLs.Elements()` after you put in two repeated elements returns an array with two repeated elements.

Nothing bad should happen, because `updateACLs` should just skip over repeated elements and the API doesn't really care if there are two different ACLs (ie, with different IDs) that are the same.

Nonetheless, I've added test cases for this and guard clauses, and added a validator for checking uniqueness (required changing from `types.Set` to `types.List`)